dealership/contracts.py

```diff
@@ -1,12 +1,75 @@
+from dealership.customers import Customer, Employee
+from dealership.vehicles import Car, Truck, Motorcycle
+
 class Contract(object):
-    pass
+    '''Represents a contract with the dealership. Must be implemented as a
+    BuyContract or a LeaseContract.'''
+    def __init__(self, vehicle, customer, num_payments):
+        self.vehicle = vehicle
+        self.customer = customer
+        self.discount_rate = 0.10
+        
+        if isinstance(self, BuyContract):
```

Comment:
It may just be my personal opinion, but I feel like the monthly payments/length should stay within their respective classes.  Passing them up to the parent through the same variable seems like it could end up being confusing, as well as being a violation of the Open Closed principle.